pdns-auth: --enable-verbose-logging (DLOG) seems to break compile in master

Commit e490ae9b1cd81107b2d0551b87672216d12022b4 seems to break compiling with "--enable-verbose-logging" (DLOG) in current master (cad0349621450e414c95b9c265b89b0f0df7cc15).
Reverting that commit seems to help (at least with compiling).

```
  CXX    packethandler.o
In file included from dnspacket.hh:54:0,
                 from packetcache.hh:35,
                 from packethandler.cc:25:
packethandler.cc: In member function ‘DNSPacket* PacketHandler::questionOrRecurse(DNSPacket*, bool*)’:
packethandler.cc:1390:199: error: operands to ?: have different types ‘const char [7]’ and ‘DNSName’
 st ANY query for '"<<target<<"', id="<<sd.domain_id<<": weDone="<<weDone<<", weHaveUnauth="<<weHaveUnauth<<", weRedirected="<<weRedirected<<", haveAlias='"<<(haveAlias.empty() ? "(none)"
                                                                                                                                                                                 ^
logger.hh:117:17: note: in definition of macro ‘DLOG’
 #define DLOG(x) x
                 ^
make[3]: *** [packethandler.o] Error 1
```

Centos 6.8, devtoolset-4

I filed a PR for this earlier today

On July 13, 2016 3:58:41 PM GMT+02:00, Andreas Jakum <notifications@github.com> wrote:
>Commit e490ae9b1cd81107b2d0551b87672216d12022b4 seems to break
>compiling with "--enable-verbose-logging" (DLOG) in current master
>(cad0349621450e414c95b9c265b89b0f0df7cc15).
>Reverting that commit seems to help (at least with compiling).
>
>
>  CXX    packethandler.o
>In file included from dnspacket.hh:54:0,
>                 from packetcache.hh:35,
>                 from packethandler.cc:25:
>packethandler.cc: In member function ‘DNSPacket*
>PacketHandler::questionOrRecurse(DNSPacket*, bool*)’:
>packethandler.cc:1390:199: error: operands to ?: have different types
>‘const char [7]’ and ‘DNSName’
>st ANY query for '"<<target<<"', id="<<sd.domain_id<<":
>weDone="<<weDone<<", weHaveUnauth="<<weHaveUnauth<<",
>weRedirected="<<weRedirected<<", haveAlias='"<<(haveAlias.empty() ?
>"(none)"
>                                                                      ^
>logger.hh:117:17: note: in definition of macro ‘DLOG’
> #define DLOG(x) x
>                 ^
>make[3]: *** [packethandler.o] Error 1
>
>
>Centos 6.8, devtoolset-4
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/PowerDNS/pdns/issues/4170

-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

--- a/pdns/packethandler.cc
+++ b/pdns/packethandler.cc
@@ -1387,7 +1387,7 @@ DNSPacket *PacketHandler::questionOrRecurse(DNSPacket *p, bool *shouldRecurse)
     }
 
 
-    DLOG(L<<"After first ANY query for '"<<target<<"', id="<<sd.domain_id<<": weDone="<<weDone<<", weHaveUnauth="<<weHaveUnauth<<", weRedirected="<<weRedirected<<", haveAlias='"<<(haveAlias.empty() ? "(none)" : haveAlias)<<"'"<<endl);
+    DLOG(L<<"After first ANY query for '"<<target<<"', id="<<sd.domain_id<<": weDone="<<weDone<<", weHaveUnauth="<<weHaveUnauth<<", weRedirected="<<weRedirected<<", haveAlias='"<<haveAlias<<"'"<<endl);
     if(p->qtype.getCode() == QType::DS && weHaveUnauth &&  !weDone && !weRedirected && d_dk.isSecuredZone(sd.qname)) {
       DLOG(L<<"Q for DS of a name for which we do have NS, but for which we don't have on a zone with DNSSEC need to provide an AUTH answer that proves we don't"<<endl);
       makeNOError(p, r, target, DNSName(), sd, 1);

--- a/pdns/tcpreceiver.cc
+++ b/pdns/tcpreceiver.cc
@@ -66,7 +66,6 @@ extern StatBag S;
 pthread_mutex_t TCPNameserver::s_plock = PTHREAD_MUTEX_INITIALIZER;
 Semaphore *TCPNameserver::d_connectionroom_sem;
 PacketHandler *TCPNameserver::s_P; 
-int TCPNameserver::s_timeout;
 NetmaskGroup TCPNameserver::d_ng;
 
 void TCPNameserver::go()
@@ -1160,7 +1159,6 @@ TCPNameserver::TCPNameserver()
 //  sem_init(&d_connectionroom_sem,0,::arg().asNum("max-tcp-connections"));
   d_connectionroom_sem = new Semaphore( ::arg().asNum( "max-tcp-connections" ));
   d_tid=0;
-  s_timeout=10;
   vector<string>locals;
   stringtok(locals,::arg()["local-address"]," ,");
 

--- a/pdns/tcpreceiver.hh
+++ b/pdns/tcpreceiver.hh
@@ -66,7 +66,6 @@ private:
 
   vector<int>d_sockets;
   vector<struct pollfd> d_prfds;
-  static int s_timeout;
 };
 
 #endif /* PDNS_TCPRECEIVER_HH */

